pinot-core/src/main/java/org/apache/pinot/core/data/manager/BaseTableDataManager.java

```diff
@@ -373,13 +372,31 @@ public void reloadSegment(String segmentName, IndexLoadingConfig indexLoadingCon
       } else {
         LOGGER.info("Reload existing segment: {} of table: {} on tier: {}", segmentName, _tableNameWithType,
             TierConfigUtils.normalizeTierName(segmentTier));
+        SegmentDirectory segmentDirectory =
+            initSegmentDirectory(segmentName, String.valueOf(zkMetadata.getCrc()), indexLoadingConfig);
+        if ((zkMetadata.getTier().equals(segmentTier))) {
```

Comment:
can inline the `ImmutableSegmentLoader.needPreprocess` to have one if-check for the case of reusing segment and the log msg in the else-branch can cover all else cases
```
if (tier is same && !ImmutableSegmentLoader.needPreprocess..) {..}
else {...}
```

Reply:
Ack